### PR TITLE
Products wishlist functionality

### PR DESCRIPTION
**Pull request by mohitdhatrak**
Deploy preview: https://deploy-preview-6--onsport.netlify.app/

Change log: Added wishlist related functionalities, no authentication related part done yet.

Screenshots: 

image

image

### CODE REVIEW

File: src/components/TopNavigation/WishlistTopNavigation.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// All module imports
+import { Link } from "react-router-dom";
+
+// All local imports
+import "./TopNavigation.css";
+import logo from "../../assets/website-logo.png";
```
Comment (ivinayakg) [inline src/components/TopNavigation/WishlistTopNavigation.jsx:6]: add all kinds of assets in the public folder , and then access it (good and a must practice), search for how to access public folder files in react if need help

File: src/context/products-context.js [hunk 2]
```diff
@@ -2,18 +2,21 @@
 import { createContext, useContext, useReducer } from "react";
 
 // All local imports
-import { productsReducerFn } from "../reducer/productsReducerFn.jsx";
+import {
+    productsInitialState,
```
Comment (ivinayakg) [inline src/context/products-context.js]: put the initial state in the context only, and then use the reducer to mutate it

File: src/pages/ProductListing/ProductListing.module.css [hunk 3]
```diff
@@ -0,0 +1,101 @@
+ul {
```
Comment (ivinayakg) [inline src/pages/ProductListing/ProductListing.module.css:1]: i love the fact you used the inbuilt module CSS functionality, great job here

File: src/pages/ProductListing/components/ProductCard/ProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                        {bestSeller ? (
+                            <span className="card-badge">Bestseller</span>
+                        ) : (
+                            ""
+                        )}
+                        <div className="card-secondary-buttons-container">
+                            <button
+                                className="card-button"
+                                onClick={(e) => {
+                                    e.stopPropagation();
```
Comment (ivinayakg) [inline src/pages/ProductListing/components/ProductCard/ProductCard.jsx:60]: define this as a function before the jsx , don't put complicated or multi liner  functions in the jsx itself , damages the code readibility

File: src/utils/displayRatingStars.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import radio from "../pages/ProductListing/components/FilterContainer/FilterContainer.module.css";
+import "../pages/ProductListing/components/ProductCard/ProductCard.css";
+
+function numOfStars(num) {
+    const arr = [];
+    for (let index = num; index > 0; index--) arr.push(index);
+    return arr;
+}
+
+export function displayRatingStars(obj) {
```
Comment (ivinayakg) [inline src/utils/displayRatingStars.jsx:11]: nice thinking of making it into a separate component

**Review by ivinayakg — APPROVED** [review]
great work!

File: src/backend/db/products.js [hunk 6]
```diff
@@ -1,30 +1,347 @@
...
-    categoryName: "horror",
-  },
-  {
-    _id: uuid(),
-    title: "Think and Grow Rich",
-    author: "Shiv Khera",
-    price: "1000",
-    categoryName: "fiction",
-  },
+    // {
```
Comment (irajbhinde) [inline src/backend/db/products.js]: You can remove the commented code

File: src/pages/ProductListing/components/FilterContainer/FilterContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import filterStyles from "./FilterContainer.module.css";
+import { filtersData, ratingFilter } from "../../../../data";
+import { CheckboxFilterType } from "../FilterTypes/CheckboxFilterType";
+import { RatingFilterType } from "../FilterTypes/RatingFilterType";
+
+export function FilterContainer() {
+    return (
+        <aside
+            aria-label="Filter options"
+            className={filterStyles["filter-container"]}
```
Comment (irajbhinde) [inline src/pages/ProductListing/components/FilterContainer/FilterContainer.jsx:11]: What's the advantage of using className the way you have and why not simply use as className="classname" ?

**Review by irajbhinde — COMMENTED** [review]
Looks good! Just look into the feedback vinayak has given

File: src/pages/ProductListing/ProductListing.module.css (hunk 3 above)
Comment (mohitdhatrak) [reply src/pages/ProductListing/ProductListing.module.css:1]: Thanks!

File: src/pages/ProductListing/components/FilterContainer/FilterContainer.jsx (hunk 7 above)
Comment (mohitdhatrak) [reply src/pages/ProductListing/components/FilterContainer/FilterContainer.jsx:11]: Actually i am using css modules here, you can read about it if you want! And ask me as well if you want to know more

File: src/backend/db/products.js (hunk 6 above)
Comment (mohitdhatrak) [reply src/backend/db/products.js]: Ok

File: src/context/products-context.js (hunk 2 above)
Comment (mohitdhatrak) [reply src/context/products-context.js]: Done

File: src/components/TopNavigation/WishlistTopNavigation.jsx (hunk 1 above)
Comment (mohitdhatrak) [reply src/components/TopNavigation/WishlistTopNavigation.jsx:6]: Will do it later after reading more! Thanks, saved this suggestion

File: src/pages/ProductListing/components/ProductCard/ProductCard.jsx (hunk 4 above)
Comment (mohitdhatrak) [reply src/pages/ProductListing/components/ProductCard/ProductCard.jsx:60]: Ok done

**Pull request merged**